ts/screens/wallet/payment/ConfirmPaymentMethodScreen.tsx

```diff
@@ -117,201 +118,192 @@ const contextualHelpMarkdown: ContextualHelpPropsMarkdown = {
-              wallet={wallet}
-              hideMenu={true}
-              hideFavoriteIcon={true}
-            />
-            <View spacer={true} />
-            {wallet.psp === undefined ? (
-              <Text>{I18n.t("payment.noPsp")}</Text>
-            ) : (
-              <Text>
-                {I18n.t("payment.currentPsp")}
-                <Text bold={true}>{` ${wallet.psp.businessName}`}</Text>
-              </Text>
-            )}
-            <TouchableDefaultOpacity onPress={this.props.pickPsp}>
-              <Text link={true} bold={true}>
-                {I18n.t("payment.changePsp")}
-              </Text>
-            </TouchableDefaultOpacity>
-            <View spacer={true} large={true} />
-            <TouchableDefaultOpacity testID="why-a-fee" onPress={this.showHelp}>
-              <Text link={true}>{I18n.t("wallet.whyAFee.title")}</Text>
-            </TouchableDefaultOpacity>
-          </View>
-        </Content>
+  const maybePsp = fromNullable(wallet.psp);
+  const fee = maybePsp.fold(undefined, psp => psp.fixedCost.amount);
 
-        <View style={styles.alert}>
-          <IconFont
-            style={styles.alertIcon}
-            name={"io-notice"}
-            size={24}
-            color={customVariables.brandDarkGray}
+  const totalAmount = maybePsp.fold(
+    verifica.importoSingoloVersamento,
+    fee =>
+      (verifica.importoSingoloVersamento as number) +
+      (fee.fixedCost.amount as number)
+  );
+
+  // the user press back during the pay web view challenge
+  const handlePayWebviewGoBack = () => {
+    Alert.alert(I18n.t("payment.abortWebView.title"), "", [
+      {
+        text: I18n.t("payment.abortWebView.confirm"),
+        onPress: () => {
+          props.dispatchCancelPayment();
+          props.dispatchEndPaymentWebview("ABORT_BY_THE_USER");
+        },
+        style: "cancel"
+      },
+      {
+        text: I18n.t("payment.abortWebView.cancel")
+      }
+    ]);
+  };
+
+  return (
+    <BaseScreenComponent
+      goBack={props.onCancel}
+      headerTitle={I18n.t("wallet.ConfirmPayment.header")}
+      contextualHelpMarkdown={contextualHelpMarkdown}
+      faqCategories={["payment"]}
+    >
+      <Content noPadded={true} bounces={false}>
+        <PaymentBannerComponent
+          currentAmount={verifica.importoSingoloVersamento}
+          paymentReason={paymentReason}
+          fee={fee as ImportoEuroCents}
+        />
+        <View style={styles.padded}>
+          <View spacer={true} />
+          <CardComponent
+            type={"Full"}
+            wallet={wallet}
+            hideMenu={true}
+            hideFavoriteIcon={true}
           />
-          <Text style={[styles.flex, styles.textColor]}>
-            <Text bold={true}>{I18n.t("global.genericAlert")}</Text>
-            {` ${I18n.t("wallet.ConfirmPayment.info")}`}
-          </Text>
+          <View spacer={true} />
+          {maybePsp.isNone() ? (
+            <Text>{I18n.t("payment.noPsp")}</Text>
```

Comment:
You can use the typography component for uniform the text style
```suggestion
            <H4 weight={"Regular"}>{I18n.t("payment.noPsp")}</H4>
            ) : (
            <H4 weight={"Regular"}>
              {I18n.t("payment.currentPsp")}
              <H4 >{` ${maybePsp.value.businessName}`}</H4>
            </H4>
          )
```